### PR TITLE
Site list functionality

### PR DESCRIPTION
**Pull request by jitendra-xoriant**
Site List functionality

### CODE REVIEW

File: src/main/java/com/mediamath/terminalone/TerminalOne.java [hunk 1]
```diff
@@ -362,6 +365,24 @@ public Strategy save(Strategy entity) throws ClientException, ParseException {
...
+   * @throws ParseException
+   *           a parse exception is thrown when the response cannot be parsed.
+   */
+  public Campaign save(Campaign entity) throws ClientException, ParseException {
+	  Campaign campaign = null;
+	    if (isAuthenticated()) {
+	    	campaign = postService.save(entity);
+	    }
+	    return campaign;
+  }
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/TerminalOne.java]: Why new `save()` for `campaign` entity? Can we try to minimize individual save methods?

File: src/main/java/com/mediamath/terminalone/models/SiteList.java [hunk 2]
```diff
@@ -140,7 +152,23 @@ public void setSitesCountApp(int sites_count_app) {
     this.sites_count_app = sites_count_app;
   }
 
-  @Override
+  public List<String> getDomains() {
+	return domains;
+}
+
+public void setDomains(List<String> domains) {
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/models/SiteList.java]: Check overall spacing and indendation

File: src/main/java/com/mediamath/terminalone/service/PostService.java [hunk 3]
```diff
@@ -174,65 +176,132 @@ private String readPostResponseToString(Response responseObj) throws ClientExcep
...
+	 *            expects a Campaign entity.
+	 * @return Strategy object.
+	 * @throws ClientException
+	 *             a client exception is thrown if any error occurs.
+	 * @throws ParseException
+	 *             a parse exception is thrown when the response cannot be
+	 *             parsed.
+	 */
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	public Campaign save(Campaign entity) throws ClientException, ParseException {
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/service/PostService.java]: Was this not covered by the generic save method previously? `TerminalOne.java` is getting long and complex I feel.

File: src/main/java/com/mediamath/terminalone/TerminalOne.java (hunk 1 above)
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/TerminalOne.java]: While Adding sitelist to Campaign, I need to insert response into campaign object, which can not be done in generic method , so separate method

File: src/main/java/com/mediamath/terminalone/models/SiteList.java (hunk 2 above)
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/models/SiteList.java]: ok

File: src/main/java/com/mediamath/terminalone/service/PostService.java (hunk 3 above)
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/service/PostService.java]: Can not use generic save method, so have to create new one

File: src/main/java/com/mediamath/terminalone/TerminalOne.java (hunk 1 above)
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/TerminalOne.java]: Got it. But can we add one task to refactor this later? I feel GET is more generic than POST and there is scope for refactoring here.

Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/TerminalOne.java]: yes we can, please add one task into Issues for POST functionality

**Pull request merged**